internal/cluster/capabilities.go

```diff
@@ -0,0 +1,45 @@
+package cluster
+
+import (
+	"encoding/json"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/internal/installcfg"
+	"github.com/pkg/errors"
+	"github.com/thoas/go-funk"
+)
+
+func GetClusterCapabilities(cluster *common.Cluster) (*installcfg.Capabilities, error) {
+	if cluster.InstallConfigOverrides == "" {
+		return nil, nil
+	}
+	var installConfig installcfg.InstallerConfigBaremetal
+	err := json.Unmarshal([]byte(cluster.InstallConfigOverrides), &installConfig)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to unmarshal install config overrides when checking capabilities for cluster %s", cluster.Name)
+	}
+	return installConfig.Capabilities, nil
+}
+
+func HasBaseCapabilities(cluster *common.Cluster, baselineCapability configv1.ClusterVersionCapabilitySet) bool {
+	capabilities, err := GetClusterCapabilities(cluster)
+	if err != nil || capabilities == nil {
+		return false
+	}
+	return capabilities.BaselineCapabilitySet == baselineCapability
+}
+
+func HasAdditionalCapabilities(cluster *common.Cluster, includes []configv1.ClusterVersionCapability) bool {
+	capabilities, err := GetClusterCapabilities(cluster)
+	if err != nil || capabilities == nil {
+		return false
+	}
+	allFound := true
```

Comment:
I think this can be simplified with [lo.Every](https://github.com/samber/lo?tab=readme-ov-file#every)` or [funk.Intersect](https://github.com/thoas/go-funk?tab=readme-ov-file#funkintersect) if you want to keep using funk. As we already have lo and funk dependency it, we can choose

Reply:
Great idea, thank you! Used lo.Every instead